### PR TITLE
Refactor CA authenticators

### PR DESCRIPTION
**Pull request by howardjohn**
This PR is split into a few commits if it makes it easier

* Removed an unused Authenticator, IDTokenAuthenticator. This is dead code that is not possible to trigger today
* Move around the different authenticators so each is in its own file
* Make the Authenticator type public and part of the the authenticate package. Its not great practice to have private structs inside public structs since they cannot be accessed
* Add assertions to all validators that they implement the interface
* Move OIDC authentication from istio_ca to the authenticate package

This is intended to have no behavioral change; only code organization. 

These pain points were found in attempting to implement a new authenticator (for a demo/proof of concept)

### CODE REVIEW

**Comment by costinm** [comment]
Which authenticator was removed ? A bit hard to follow - the cleanup is welcomed. 

Keep in mind that 'dead code' is in part due to unfinished implementation - the GCP/AWS authenticators will probably be needed even if they're now disabled.

**Comment by costinm** [comment]
If I read this correctly - you removed the OIDC token verifier that had google hardcoded values, but kept the genric one. I think that's right - for GCP VMs we should use the generic OIDC, no need to hardcode GCP.

**Comment by howardjohn** [comment]
Yeah exactly, removing the hardcoded google oidc and moving the generic oidc into the same package with the other authenticators

File: security/pkg/server/ca/authenticate/kube_jwt.go [hunk 1]
```diff
@@ -45,6 +47,8 @@ type KubeJWTAuthenticator struct {
 	remoteKubeClientGetter RemoteKubeClientGetter
 }
 
+var _ Authenticator = &KubeJWTAuthenticator{}
```
[the remark this answers is not in the document]
Comment (howardjohn) [reply security/pkg/server/ca/authenticate/kube_jwt.go:50]: This asserts that it implements the Authenticator interface. See https://stackoverflow.com/questions/44976862/what-is-an-interface-assertion for more info.

[the remark this answers is not in the document]
Comment (myidpt) [reply security/pkg/server/ca/authenticate/kube_jwt.go:50]: I see. Thanks.
